### PR TITLE
Explicit support for deprecated directive in the schema

### PR DESCRIPTION
**Pull request by dkuc84**
Resolves https://github.com/ExpediaDotCom/graphql-kotlin/issues/246

### CODE REVIEW

**Comment by dkuc84** [comment]
`GraphQLSchema.print()` extension prints the following
Screen Shot 2019-07-01 at 3 50 58 PM

File: src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchema.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+private val schemaPrinter = SchemaPrinter(
+    SchemaPrinter.Options.defaultOptions()
+        .includeScalarTypes(true)
+        .includeExtendedScalarTypes(true)
+        .includeIntrospectionTypes(true)
+        .includeSchemaDefintion(true)
+        .includeDirectives(false)
+)
+
+fun GraphQLSchema.print(): String {
```
Comment (smyrick) [inline src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchema.kt]: Do we want to add the config options or accept a optional `SchemaPrinter` as input to `print()`?

Comment (dkuc84) [reply src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchema.kt]: sure, can expose the boolean flags

File: src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchemaExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    includeSchemaDefinition: Boolean = true,
+    includeDirectives: Boolean = true
+): String {
+    val schemaPrinter = SchemaPrinter(
+        SchemaPrinter.Options.defaultOptions()
+            .includeScalarTypes(includeScalarTypes)
+            .includeExtendedScalarTypes(includeExtendedScalarTypes)
+            .includeIntrospectionTypes(includeIntrospectionTypes)
+            .includeSchemaDefintion(includeSchemaDefinition)
+            .includeDirectives(false)
```
Comment (dkuc84) [inline src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchemaExtensions.kt]: ugh.... this flag removes both directive definitions as well as their target locations :(

Comment (smyrick) [reply src/main/kotlin/com/expedia/graphql/extensions/GraphQLSchemaExtensions.kt]: Added an issue here: https://github.com/graphql-java/graphql-java/issues/1587

**Pull request merged**